string/test/string_test.cpp

```diff
@@ -0,0 +1,91 @@
+#include <cstdio>
+#include <string.h>
+#include "string/string.h"
+
+#pragma clang diagnostic push
```

Comment:
not sure about leaving this in, I'm fine with it for now but technically we only support GCC. Some of the scheduler code uses non-C++ standard features that I'm not sure are in Clang so we're not really cross-compiler at the moment.

Reply:
on second thought it probably doesn't matter because GCC will just complain about an unknown pragma and probably still compile. If you can just check that it compiles with GCC before we merge that would be good tho.